spring-integration-zeromq/src/main/java/org/springframework/integration/zeromq/outbound/ZeroMqMessageHandler.java

```diff
@@ -244,7 +255,12 @@ protected Mono<Void> handleMessageInternal(Message<?> message) {
 						if (socket.base() instanceof Pub) {
 							String topic = this.topicExpression.getValue(this.evaluationContext, message, String.class);
 							if (topic != null) {
-								msg.wrap(new ZFrame(topic));
+								var frame = new ZFrame(topic);
+								if (wrapTopic) {
```

Comment:
`this.wrapTopic`.
And apparently the `ZeroMqMessageProducer` has to be fixed opposite way for its:
```
headers = Collections.singletonMap(ZeroMqHeaders.TOPIC, msg.unwrap().getString(ZMQ.CHARSET));
```
logic.